### DISCUSSION TITLE
Path patterns not working

### DISCUSSION THREAD

**Discussion by HAEKADI** [community/community#25638] [Actions]
<p>Hello,</p>
<p>I used the <code>tests/**/*.py</code> syntax to match all the Python files within the <code>tests</code> directory, as specified <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-file-paths" rel="noopener nofollow ugc">here</a>.</p>
<p>However, it only matches the files within the first subdirectory and completely ignores the Python files in the current directory and the sub-subdirectories.</p>
<pre><code class="lang-auto">  test-python:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: [ 3.7, 3.8, 3.9 ]
    needs: build
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}
      - name: Cache build
        id: restore-build
        uses: actions/cache@v2
        with:
          path: ${{ env.pythonLocation }}
          key: ${{ env.pythonLocation }}-build-${{ hashFiles('setup.py') }}
      - run: ls -d tests/**/*.py
      - run: openfisca test tests/**/*.py
</code></pre>
<p>Locally, it matched the behavior of <code>tests/*/*.py</code>.</p>

**Comment by airtower-luna** [comment]
<p>To match any number of directories with <code>**</code> in Bash, you need to activate the “globstar” option:</p>
<pre><code class="lang-auto">shopt -s globstar
</code></pre>
<p>You could put that into your <code>run</code> commands, or use the <code>shell</code> parameter for the stop and the <code>-O</code> option of Bash to activate it on shell start.</p>

**Comment by HAEKADI** [comment]
<p>I have already tried <code>shopt -s globstar</code> but it didn’t change anything.</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/5/a532555b4a060ec68b5002efde23b4fc39155af1.png" data-download-href="/uploads/short-url/nzoCx01hJYSlskDn2ImqVGsxubn.png?dl=1" title="Screenshot 2021-08-30 at 14.08.09" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/48837850/181092309-3885b158-cbc9-4f5f-a94c-6b4ee142dfdf.png" alt="Screenshot 2021-08-30 at 14.08.09" data-base62-sha1="nzoCx01hJYSlskDn2ImqVGsxubn" width="615" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a532555b4a060ec68b5002efde23b4fc39155af1_2_615x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a532555b4a060ec68b5002efde23b4fc39155af1_2_922x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/5/a532555b4a060ec68b5002efde23b4fc39155af1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a532555b4a060ec68b5002efde23b4fc39155af1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-08-30 at 14.08.09</span><span class="informations">1229×999 114 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
I still have missing files.

**Comment by airtower-luna** [comment, marked as answer]
<p>Each step gets a fresh shell, so changes to shell settings are lost after the step. You need to set the globstar option <em>in the step</em> where you want to use the <code>**</code> glob.</p>
<p>Try:</p>
<pre><code class="lang-yaml">      - run: |
          shopt -s globstar
          ls tests/**/*.py
</code></pre>

**Comment by HAEKADI** [comment]
<p>This works. Thank you!</p>
